### PR TITLE
Enable soft deletes by default for new tables

### PR DESCRIPTION
**Pull request by mfussenegger**
## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by kovrus** [comment]
my fault https://github.com/crate/crate/blob/master/server/src/test/java/io/crate/integrationtests/OpenCloseTableIntegrationTest.java#L119 , we have to do assert `uncommittedTranslogOps` instead of `numberOfDocs` if soft deletes are enabled.

**Review by kovrus — APPROVED** [review]
lgtm !!

File: docs/sql/statements/create-table.rst [hunk 1]
```diff
@@ -417,6 +417,30 @@ Allows to have a read only table that additionally can be deleted.
...
+``soft_deletes.enabled``
+------------------------
+
+Indicates whether soft deletes are enabled or disabled. 
+
+Soft deletes allow CrateDB to preserve recent deletions within the Lucene
+index. This information is used for shard recovery.
+
+Before the introduction of soft deletes, CrateDB had to retain the information
+in the Translog. Using soft deletes uses less storage than the Translog
```
Comment (mechanomi) [inline docs/sql/statements/create-table.rst:431]: might be beneficial to link "Translog" to https://crate.io/docs/crate/reference/en/4.2/concepts/storage-consistency.html#durability

**Review by mechanomi — COMMENTED** [review]
left one comment. but lgtm

**Pull request merged**